pennylane/gradients/__init__.py

```diff
@@ -11,7 +11,266 @@
+        [ 0.04673668,  0.09442394,  0.14409127]], requires_grad=True)
+
+.. note::
+
+    If your circuit contains any operations not supported by the gradient
+    transform, the transform will attempt to automatically decompose the
+    circuit into only operations that support gradients.
+
+.. note::
+
+    If you wish to only return the purely **quantum** component of the
+    gradient---that is, the gradient of the output with respect to
+    **gate** arguments, not QNode arguments---pass ``hybrid=False``
+    when applying the transform:
+
+    >>> qml.gradients.param_shift(circuit, hybrid=False)(weights)
+
+
+Differentiating gradient transforms
+-----------------------------------
+
+Gradient transforms are themselves differentiable, allowing higher-order
+gradients to be computed:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.qnode(dev)
+    def circuit(weights):
+        qml.RX(weights[0], wires=0)
+        qml.RY(weights[1], wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.RX(weights[2], wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+>>> weights = np.array([0.1, 0.2, 0.3], requires_grad=True)
+>>> circuit(weights)
+tensor(0.9316158, requires_grad=True)
+>>> qml.gradients.param_shift(circuit)(weights)  # gradient
+array([[-0.09347337, -0.18884787, -0.28818254]])
+>>> qml.jacobian(qml.gradients.param_shift(circuit))(weights)  # hessian
+array([[[-0.9316158 ,  0.01894799,  0.0289147 ],
+        [ 0.01894799, -0.9316158 ,  0.05841749],
+        [ 0.0289147 ,  0.05841749, -0.9316158 ]]])
+
+
+Transforming tapes
+------------------
+
+Gradient transforms can be applied to low-level :class:`~.QuantumTape` objects,
+a datastructure representing variational quantum algorithms:
+
+.. code-block:: python
+
+    weights = np.array([0.1, 0.2, 0.3], requires_grad=True)
+
+    with qml.tape.JacobianTape() as tape:
+        qml.RX(weights[0], wires=0)
+        qml.RY(weights[1], wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.RX(weights[2], wires=1)
+        qml.expval(qml.PauliZ(1))
+
+Unlike when transforming a QNode, transforming a tape directly
+will perform no implicit quantum device evaluation. Instead, it returns
+the processed tapes, and a post-processing function, which together
+define the gradient:
+
+>>> gradient_tapes, fn = qml.gradients.param_shift(tape)
+>>> gradient_tapes
+[<JacobianTape: wires=[0, 1], params=3>,
+ <JacobianTape: wires=[0, 1], params=3>,
+ <JacobianTape: wires=[0, 1], params=3>,
+ <JacobianTape: wires=[0, 1], params=3>,
+ <JacobianTape: wires=[0, 1], params=3>,
+ <JacobianTape: wires=[0, 1], params=3>]
+
+This can be useful if the underlying circuits representing the gradient
+computation need to be analyzed.
+
+The output tapes can then be evaluated and post-processed to retrieve
+the gradient:
+
+>>> dev = qml.device("default.qubit", wires=2)
+>>> from pennylane.interfaces.batch import execute
+>>> fn(execute(gradient_tapes, dev, None))

```

Comment:
Would it be worth including a little detail  above on how to use the post processing function? E.g what type of arg it expects.

Reply:
Good suggestion, thanks @anthayes92!